src/pages/SupplierInvoicePage.js

```diff
@@ -107,56 +208,234 @@ export class SupplierInvoicePage extends GenericPage {
           title: `${pageInfoStrings.confirm_date}:`,
           info: formatDate(transaction.confirmDate),
         },
+        {
+          title: `${pageInfoStrings.total_price}:`,
+          info: this.state.totalPrice,
+        },
       ],
       [
         {
+          title: `${pageInfoStrings.supplier}:`,
+          info: this.props.transaction.otherParty && this.props.transaction.otherParty.name,
+        },
+        {
           title: `${pageInfoStrings.their_ref}:`,
           info: transaction.theirRef,
+          onPress: this.openTheirRefEditor,
+          editableType: 'text',
         },
         {
           title: `${pageInfoStrings.comment}:`,
           info: transaction.comment,
+          onPress: this.openCommentEditor,
+          editableType: 'text',
         },
       ],
     ];
-    return <PageInfo columns={infoColumns} />;
+    return (
+      <PageInfo columns={infoColumns} isEditingDisabled={this.props.transaction.isFinalised} />
+    );
   }
 
-  renderCell(key, transactionItem) {
+  renderCell(key, transactionBatch) {
+    const isEditable = !this.props.transaction.isFinalised;
+    const type = isEditable ? 'editable' : 'text';
+    const editableCell = {
+      type: type,
```

Comment:
@Chris-Petty , this is a shared SupplierInvoicePage now, I was talking about SupplierInvoice not created but received.  So when Supplier Invoice comes from central store:

<img width="961" alt="screen shot 2017-08-03 at 2 18 11 pm" src="https://user-images.githubusercontent.com/26194949/28903404-c5048f74-7858-11e7-8940-a7cd3580fcb2.png">

Editable fields: Their Ref, Comment, pack size, quantity, batch, expiry, cost price, delete.. and can add a new line.. It looks to me like it's adding room for error. @craigdrown  ?